src/tir/schedule/primitive/cache_read_write.cc

```diff
@@ -75,8 +77,8 @@ struct CacheStageInfo {
   Stmt cache_stage;
   /*! \brief The map used for ScheduleStateNode::Replace. */
   Map<Block, Block> block_reuse;
-  /*! \brief A list of blocks that will consume the new cache. */
-  Array<StmtSRef> consumer_blocks;
+  /*! \brief A set of blocks that will consume the new cache. */
+  std::unordered_set<StmtSRef, ObjectHash, ObjectEqual> consumer_blocks;
```

Comment:
I was curious if the ordering of `consumer_blocks` might matter, and would it lead to undeterministic behavior?

Reply:
I checked the uses of `consumer_blocks`, the order doesn't matter.